plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/LocalizationPlugin.java

```diff
@@ -0,0 +1,114 @@
+package com.mapbox.mapboxsdk.plugins.testapp.activity;
+
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * The localization plugin enables localization of map labels into the user’s preferred language
+ * <p>
+ * Initialise this plugin in the {@link com.mapbox.mapboxsdk.maps.OnMapReadyCallback#onMapReady(MapboxMap)} and provide
+ * a valid instance of a {@link MapboxMap}.
+ * </p>
+ */
+public final class LocalizationPlugin {
+
+  private List<Layer> listOfMapLayers;
+
+  /**
+   * Create a localization plugin.
+   *
+   * @param mapboxMap the MapboxMap to apply localization plugin with
+   * @since 0.1.0
+   */
+  public LocalizationPlugin(@NonNull final MapboxMap mapboxMap) {
+    listOfMapLayers = mapboxMap.getLayers();
+    String deviceLanguage = Locale.getDefault().getLanguage();
```

Comment:
According to [this documentation](https://developer.android.com/reference/java/util/Locale.html), `getLanguage()` returns deprecated ISO 639 codes, such as `iw` for Hebrew instead of the current standard `he`.

Additionally, will this method return `zh-Hans` when the user chooses Chinese (Simplified) in their device settings?

Reply:
https://gist.github.com/SeanZoR/9152167 👀

Reply:
https://developer.android.com/reference/java/util/Locale.html


<img width="1024" alt="screen shot 2018-01-04 at 4 26 10 pm" src="https://user-images.githubusercontent.com/4394910/34590496-00b16524-f16c-11e7-8836-b34c4c52a508.png">

Reply:
https://www.loc.gov/standards/iso639-2/php/code_changes.php

Reply:
https://web.archive.org/web/20131209133616/http://code.google.com/p/android/issues/detail?id=3639

Reply:
I made [this commit](https://github.com/mapbox/mapbox-plugins-android/pull/74/commits/fffe2d7ee81d90748cdb9867bd55a6eff115af2a) which checks for and adjusts known deprecated ISO 639 codes. After lots of research, it doesn't seem like there's a single method that covers the backward and forwards compatibility that @1ec5 mentions above. Let's let my commit ✅  this comment for now so that this isn't a blocker for getting this plugin released.

Reply:
[This commit](https://github.com/mapbox/mapbox-plugins-android/pull/74/commits/35618ce7a43b0d8960534967246e695636811984) addresses simplified/traditional Chinese selection

Reply:
How about [toLanguageTag](https://developer.android.com/reference/java/util/Locale.html#toLanguageTag())?

Reply:
yea, looked into it. For some reason, I think I read that it wouldn't work both ways, but I don't remember exactly why now 😵 . That link you just gave above, _does_ sound like it would work for covering back and forwards compatibility. Will look back into it

Reply:
Using `toLanguageTag()` if api level is above 21.

Both issues brought up in OP have been addressed.